src/main/java/sirius/db/jdbc/GuardedUpdateQuery.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.db.jdbc;
+
+import sirius.db.mixing.EntityDescriptor;
+import sirius.db.mixing.Mapping;
+import sirius.kernel.commons.Watch;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Provides a simple helper to generate efficient update queries on {@link SQLEntity entities}.
+ * <p>
+ * <b>Note that this will not execute any {@link sirius.db.mixing.annotations.BeforeSave} handlers.</b>
+ * <p>
+ * This can be used to generate queries like {@code UPDATE table SET x = 'Value' WHERE y = 'Value'} without having
+ * to worry about typing field and table names correctly. Such update are sometimes more efficient as a lot of the
+ * framework overhead is reduced (this is essentially just a builder which generates a {@link PreparedStatement}).
+ * This can also be used for conditional updates (e.g. optimistic locking algorithms and the like.
+ * <p>
+ * Note that this, being a builder class with minimal state overhead, does not support to add another update
+ * via {@link #set(Mapping, Object)} once the first constraint has been added via {@link #where(Mapping, Object)}.
+ */
+public class GuardedUpdateQuery {
+
+    private static final String MICROTIMING_KEY = "SQL-GUARDED-UPDATE";
+    private final EntityDescriptor descriptor;
+    private final Database db;
+    private StringBuilder queryBuilder = new StringBuilder();
+    private boolean wherePart;
+    private boolean setPart;
```

Comment:
Use Monoflops instead of booleans to increase readability

(or simply rename the booleans to wherePartStarted, setPartStarted)